crates/rpc/rpc-eth-api/src/helpers/estimate.rs

```diff
@@ -330,32 +327,11 @@ pub fn update_estimated_gas_range(
             // Cap the highest gas limit with the succeeding gas limit.
             *highest_gas_limit = tx_gas_limit;
         }
-        ExecutionResult::Revert { .. } => {
-            // Increase the lowest gas limit.
+        ExecutionResult::Revert { .. } | ExecutionResult::Halt { .. } => {
+            // We know that transaction succeeded with a higher gas limit before, so any failure
+            // means that we need to increase it.
```

Comment:
hmm, this now exempts all haltreasons, I am not sure if this is okay.
judging from the docs it might because we don't expect those.
so this seems fine, 
I'd like to keep the docs+context here